File: ChatGPT_Bandit/78_1c.py
import subprocess
import sys

def run_id_command(username):
    try:
        # Run the id command
        result = subprocess.run(['id', username], capture_output=True, text=True)
        # Print the output of the command
        print(result.stdout)
        # Return the exit status
        return result.returncode
    except Exception as e:
        print(f"An error occurred: {e}")
        return 1

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python script.py <username>")
        sys.exit(1)
    
    username = sys.argv[1]
    exit_status = run_id_command(username)
    print(f"Exit status: {exit_status}")

File: GithubCopilot_CPP/CWE-78/78_1c.cpp
#include <iostream>
#include <cstdlib>
#include <string>

int runCommand(const std::string& username) {
    std::string command = "id " + username;
    int exitStatus = system(command.c_str());
    return exitStatus;
}

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <username>" << std::endl;
        return 1;
    }

    std::string username = argv[1];
    int status = runCommand(username);

    std::cout << "Exit status: " << status << std::endl;

    return status;
}